### ISSUE TITLE
MySQL Closes connection

### ISSUE THREAD

**Issue by Mowstyl** [ProjectKorra/ProjectKorra#867]
When a server is empty for a long time, the DB server can timeout the connection. If someone enters after that has happened, the plugin doesn't work and prints huge messages. I'll attach the file GeneralMethods.java with that problem solved (I'm not sure, I haven't tested it).
The only change is on the first lines of the method "createBendingPlayerAsynchronously".

[GeneralMethods.zip](https://github.com/ProjectKorra/ProjectKorra/files/1435872/GeneralMethods.zip)

I hope it can help. It's a great plugin and I want it to become greater, I know you can do it 👍!

**Comment by Mowstyl** [comment]
This is the huge message I was talking about:
[18:18:39] [LoginProcessingThread/INFO]: UUID of player Mowstyl is ********-****-****-****-************
[18:18:40] [Server thread/INFO]: Mowstyl[/***.***.***.***:37369] logged in with entity id 132402 at ([JhoSPVP]4344.254247923307, 67.0, 2505.4008915866357)
[18:18:40] [Server thread/WARN]: com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException: No operations allowed after connection closed.
[18:18:40] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[18:18:40] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[18:18:40] [Server thread/WARN]: 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[18:18:40] [Server thread/WARN]: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.Util.getInstance(Util.java:408)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:918)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:897)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:886)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:860)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.ConnectionImpl.throwConnectionClosedException(ConnectionImpl.java:1187)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.ConnectionImpl.checkClosed(ConnectionImpl.java:1182)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.ConnectionImpl.prepareStatement(ConnectionImpl.java:4071)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.ConnectionImpl.prepareStatement(ConnectionImpl.java:4040)
[18:18:40] [Server thread/WARN]: 	at com.projectkorra.projectkorra.storage.Database.readQuery(Database.java:119)
[18:18:40] [Server thread/WARN]: 	at com.projectkorra.projectkorra.GeneralMethods.createBendingPlayerAsynchronously(GeneralMethods.java:304)
[18:18:40] [Server thread/WARN]: 	at com.projectkorra.projectkorra.GeneralMethods.access$000(GeneralMethods.java:143)
[18:18:40] [Server thread/WARN]: 	at com.projectkorra.projectkorra.GeneralMethods$1.run(GeneralMethods.java:297)
[18:18:40] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:58)
[18:18:40] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:356)
[18:18:40] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:833)
[18:18:40] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426)
[18:18:40] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767)
[18:18:40] [Server thread/WARN]: 	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665)
[18:18:40] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[18:18:40] [Server thread/WARN]: Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: The last packet successfully received from the server was 1,474,113,827 milliseconds ago.  The last packet sent successfully to the server was 1,474,113,827 milliseconds ago. is longer than the server configured value of 'wait_timeout'. You should consider either expiring and/or testing connection validity before use in your application, increasing the server configured values for client timeouts, or using the Connector/J connection property 'autoReconnect=true' to avoid this problem.
[18:18:40] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[18:18:40] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[18:18:40] [Server thread/WARN]: 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[18:18:40] [Server thread/WARN]: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.MysqlIO.send(MysqlIO.java:3746)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2509)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2487)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1858)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:1966)
[18:18:40] [Server thread/WARN]: 	at com.projectkorra.projectkorra.storage.Database.readQuery(Database.java:120)
[18:18:40] [Server thread/WARN]: 	... 10 more
[18:18:40] [Server thread/WARN]: Caused by: java.net.SocketException: Broken pipe (Write failed)
[18:18:40] [Server thread/WARN]: 	at java.net.SocketOutputStream.socketWrite0(Native Method)
[18:18:40] [Server thread/WARN]: 	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111)
[18:18:40] [Server thread/WARN]: 	at java.net.SocketOutputStream.write(SocketOutputStream.java:155)
[18:18:40] [Server thread/WARN]: 	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
[18:18:40] [Server thread/WARN]: 	at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
[18:18:40] [Server thread/WARN]: 	at com.mysql.jdbc.MysqlIO.send(MysqlIO.java:3728)
[18:18:40] [Server thread/WARN]: 	... 16 more
[18:18:40] [Server thread/WARN]: [ProjectKorra] Task #10253515 for ProjectKorra v1.8.6 generated an exception
java.lang.NullPointerException: null
	at com.projectkorra.projectkorra.GeneralMethods.createBendingPlayerAsynchronously(GeneralMethods.java:306) ~[?:?]
	at com.projectkorra.projectkorra.GeneralMethods.access$000(GeneralMethods.java:143) ~[?:?]
	at com.projectkorra.projectkorra.GeneralMethods$1.run(GeneralMethods.java:297) ~[?:?]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:58) ~[paperclip.jar:git-Paper-1238]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:356) [paperclip.jar:git-Paper-1238]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:833) [paperclip.jar:git-Paper-1238]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [paperclip.jar:git-Paper-1238]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [paperclip.jar:git-Paper-1238]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [paperclip.jar:git-Paper-1238]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]

The message starts spamming the last part after that.

**Comment by PhanaticD** [comment]
which version are you using, I think I might have fixed this in the WIP version here https://github.com/ProjectKorra/ProjectKorra/commit/29999942b26e59a2ba391abf28eec8171f45cc7c

**Comment by Mowstyl** [comment]
1.8.6. I downloaded it before that commit. Looking the new code I think the error I have will be fixed.
I still think the select in GeneralMethods should be inside the try-catch because it can also thow an SQLException and handling the exception is way more elegant than just letting the thread crash if something bad happens.
Thank you for your quick answer and forgive me for not looking at the WIP branch to see if the error has already been solved.
